commands/http/handler.go

```diff
@@ -121,6 +121,16 @@ func (i internalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	i.ctx.Context = ctx
 	req.SetContext(i.ctx)
 
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func() {
+			select {
+			case <-cn.CloseNotify():
+				cancel()
+			case <-ctx.Done():
+			}
+		}()
+	}
+
```

Comment:
ok i think i found the problem:
- this registers a canceller goroutine.
- `i.root.Cal(req)` runs, issuing the http request
- http request returns, and closes _normally_.
- before the post-call stuff (writing out result) happens,
- the CloseNotifier signals to cancel the request.

Soooo i think we need this inside (the already complicated):

```go
func (c *client) Send(req cmds.Request) (cmds.Response, error)
```